crates/node/service/src/service/mode.rs

```diff
@@ -14,6 +14,21 @@ pub enum NodeMode {
     Sequencer,
 }
 
+impl std::str::FromStr for NodeMode {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "validator" => Ok(Self::Validator),
+            "sequencer" => Ok(Self::Sequencer),
+            _ => Err(format!(
+                "Invalid node mode: '{}'. Valid options are: 'validator', 'sequencer'",
+                s
+            )),
+        }
+    }
+}
```

Comment:
This should be a derive more [`FromStr`](https://docs.rs/derive_more/latest/derive_more/derive.FromStr.html)